ksqldb-engine/src/main/java/io/confluent/ksql/schema/ksql/inference/SchemaRegisterInjector.java

```diff
@@ -102,8 +105,15 @@ private void registerForCreateSource(final ConfiguredStatement<? extends CreateS
         cs.getSessionConfig().getConfig(false)
     );
 
+    final SchemaAndId rawKeySchema = (SchemaAndId) cs.getSessionConfig().getOverrides()
+        .get(CommonCreateConfigs.KEY_SCHEMA_ID);
```

Comment:
I saw a different code in `DefaultSchemaInjector.java` where `KEY_SCHEMA_ID` and `VALUE_SCHEMA_ID` are strings.
```
final String schemaIdName =
            isKey ? CommonCreateConfigs.KEY_SCHEMA_ID : CommonCreateConfigs.VALUE_SCHEMA_ID;
```

Why is the casting to `SchemaAndId`? Doesn't it fail?

Reply:
This is casting value to `SchemaAndId` type: `.get(CommonCreateConfigs.KEY_SCHEMA_ID);`

Reply:
`SchemaAndId` was put to `SessionConfig.overrides` here: https://github.com/confluentinc/ksql/blob/master/ksqldb-engine/src/main/java/io/confluent/ksql/schema/ksql/inference/DefaultSchemaInjector.java#L120-L128

It's not put to `CreateSourceProperties`